### PR TITLE
fix(2432): support external trigger format for same pipeline

### PR DESCRIPTION
**Pull request by jithine**
## Context

Users can use external trigger syntax for same pipeline 🙄 

## References

https://github.com/screwdriver-cd/screwdriver/issues/2432

## License
I confirm that this contribution is made under a BSD license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: plugins/builds/index.js [hunk 1]
```diff
@@ -618,6 +620,10 @@ async function createJoinObject(nextJobs, current, eventFactory) {
         const jobInfo = getPipelineAndJob(jobName, current.pipeline.id);
         const pid = jobInfo.externalPipelineId;
         const jName = jobInfo.externalJobName;
+        // needed to determine if join in same pipeline
+        // is expressed in external format
+        const isExternal = jobInfo.isExternal;
```
Comment (tkyi) [inline plugins/builds/index.js:625]:
```suggestion
        // needed to determine if join in same pipeline
        // is expressed in external format
         const { externalPipelineId: pid, externalJobName: jName, isExternal } = jobInfo;
```

File: plugins/builds/index.js [hunk 2]
```diff
@@ -961,23 +967,36 @@ const buildsPlugin = {
...
+                const isCurrentPipeline = +pid === current.pipeline.id;
+
+                if (isCurrentPipeline) {
                     for (const nextJobName of Object.keys(pipelineJoinData[pid].jobs)) {
                         try {
-                            const resource = `pipeline:${current.pipeline.id}:event:${current.event.id}`;
-                            const lock = await locker.lock(resource);
+                            const forceExternalTrigger = pipelineJoinData[pid].jobs[nextJobName].isExternal;
+
+                            triggerCurrentPipelineAsExternal = triggerCurrentPipelineAsExternal || forceExternalTrigger;
```
Comment (tkyi) [inline plugins/builds/index.js:978]: should order be switched?
```suggestion
                            triggerCurrentPipelineAsExternal = forceExternalTrigger || triggerCurrentPipelineAsExternal;
```

File: plugins/builds/index.js [hunk 3]
```diff
@@ -961,23 +967,36 @@ const buildsPlugin = {
...
+                            if (!forceExternalTrigger) {
```
Comment (tkyi) [inline plugins/builds/index.js:979]:
```suggestion
                            if (!triggerCurrentPipelineAsExternal) {
```

File: plugins/builds/index.js (hunk 2 above)
Comment (jithine) [reply plugins/builds/index.js:978]: doesn't matter, we are looking for atleast one job which is of external format for same pipeline

File: plugins/builds/index.js (hunk 3 above)
Comment (jithine) [reply plugins/builds/index.js:979]: no this won't be correct, there could be mix of jobs, some in external format and some not. we are looking specifically at current job

**Review by klu909 — APPROVED** [review]

**Pull request merged**